### ISSUE TITLE
beit_large_patch16_224 does not show expected performance

### ISSUE THREAD

**Issue by bjfranks** [microsoft/unilm#519]
**Describe the bug**
Model I am using (UniLM, MiniLM, LayoutLM ...):
BEiT-L. Or more specifically the weights in beit_large_patch16_224_pt22k_ft22kto1k.pth

The problem arises when using:
* [ ] the official example scripts: (give details below)
* [X] my own modified scripts: (give details below)

A clear and concise description of what the bug is:
When I load the weights in beit_large_patch16_224_pt22k_ft22kto1k.pth and calculate the accuracy on ImageNet1k-val I cannot observer an accuracy@1 of 88.60%.

**To Reproduce**
1. I am using timm==0.4.12, and thus not the metnioned timm==0.3.2. This is due to me needing the newer PyTorch versions and timm==0.3.2 is not compatible with my version. It produces a known error.
2. 
```python
from timm.models import create_model
import modeling_finetune

model = create_model(
        'beit_large_patch16_224',
        pretrained=False,
        num_classes=1000,
        drop_rate=0.0,
        drop_path_rate=0.2,
        attn_drop_rate=0.0,
        drop_block_rate=None,
        use_mean_pooling=False,
        init_scale=0.001,
        use_rel_pos_bias=True,
        use_abs_pos_emb=False,
        init_values=1e-5,
    )
state_dict = torch.load('beit_large_patch16_224_pt22k_ft22kto1k.pth')
state_dict['model']['norm.weight'] = state_dict['model'].pop('fc_norm.weight')
state_dict['model']['norm.bias'] = state_dict['model'].pop('fc_norm.bias')
model.load_state_dict(state_dict['model'])
```
I suspect that due to the newer timm version some naming was changed which requires me to load the weights of the fc_norm layer in this slightly changed fashion.

**Expected behavior**
If the model is loaded as above I would expect the accuracy to be as reported. I.e. around 88.6.

- Platform: Linux
- Python version: 3.7.3
- PyTorch version (GPU?): 1.10.0

**Comment by addf400** [comment]
@bjfranks We recommend following our [guidelines](https://github.com/microsoft/unilm/tree/master/beit#setup) to set the environment (or using recommend docker) without pain. 

`beit_large_patch16_224_pt22k_ft22kto1k` is fine-tuned by using mean-pooling. For correctness we need to ensure that the inference is the same as the training, please set `use_mean_pooling=False` and do not need to adjust the state_dict.

**Comment by bjfranks** [comment]
@addf400 That was apprently the only issue. I now get 0.87476@1 0.98304@5, which I guess is rather close to the expected value. I wonder what model exactly gets the 0.886@1 and 0.9866@5 reported on https://github.com/microsoft/unilm/tree/master/beit. Thanks and your code was quite easily usable in another projekt, even though you could explain the use of your pretrained weights a bit better.

**Comment by addf400** [comment]
> @addf400 That was apprently the only issue. I now get 0.87476@1 0.98304@5, which I guess is rather close to the expected value. I wonder what model exactly gets the 0.886@1 and 0.9866@5 reported on https://github.com/microsoft/unilm/tree/master/beit. Thanks and your code was quite easily usable in another projekt, even though you could explain the use of your pretrained weights a bit better.

Evaluate our fine-tuned BEiT-large model in 512 resolution on ImageNet val with a single GPU:
```bash
python run_class_finetuning.py \
    --eval --model beit_large_patch16_512 --input_size 512 --data_path /path/to/imagenet \
    --resume https://unilm.blob.core.windows.net/beit/beit_large_patch16_512_pt22k_ft22kto1k.pth 
```
Expected results:

* Acc@1 88.600 Acc@5 98.658 loss 0.474
